tests/scripts/depends.py

```diff
@@ -387,15 +387,23 @@ def handle_exclusive_groups(config_settings, symbol):
 
 def turn_off_dependencies(config_settings):
     """For every option turned off config_settings, also turn off what depends on it.
-An option O is turned off if config_settings[O] is False."""
+
+    An option O is turned off if config_settings[O] is False.
+    Handle the dependencies recursively.
+    """
     for key, value in sorted(config_settings.items()):
         if value is not False:
             continue
+
+        # Save the processed settings to handle cross referencies
         revdep = set(REVERSE_DEPENDENCIES.get(key, []))
+        history = set()
         while revdep:
             dep = revdep.pop()
+            history.add(dep)
             config_settings[dep] = False
-            revdep.update(REVERSE_DEPENDENCIES.get(dep, []))
+            # Do not add symbols which are already processed
+            revdep.update(set(REVERSE_DEPENDENCIES.get(dep, [])) - history)
```

Comment:
We are updating an iterable while parsing it. That is not recommended in Python unless something has changed in later revisions? 

From a quick look sets are implemented in a way that could consistently work, but I could make a copy just in case.